ihmc-high-level-behaviors/src/main/java/us/ihmc/behaviors/behaviorTree/ros2/ROS2BehaviorTreeSubscription.java

```diff
@@ -48,13 +49,23 @@ public ROS2BehaviorTreeSubscription(BehaviorTreeState behaviorTreeState,
             messageRecievedCallback.run();
          }
 
-         long nextSequenceID = behaviorTreeStateMessage.getSequenceId();
-         if (previousSequenceID > -1)
+         long receivedSequenceID = behaviorTreeStateMessage.getSequenceId();
+         long expectedSequenceID = previousSequenceID + 1;
+         long difference = receivedSequenceID - expectedSequenceID;
+
+         if (Math.abs(difference) < ROS2BehaviorTreeState.SYNC_FREQUENCY) // Account for restarts
          {
-            long expectedSequenceID = previousSequenceID + 1;
-            messageDropCount += nextSequenceID - expectedSequenceID;
+            if (difference > 0)
+            {
+               messageDropCount += difference;
+            }
+            else if (difference < 0)
+            {
+               outOfOrderCount -= difference;
+            }
```

Comment:
Calculate dropped messages and out of order messages better here. still might be flawed, but seems good enough to see when things are going wrong